app/src/main/java/org/schabi/newpipe/local/feed/FeedFragment.kt

```diff
@@ -69,8 +89,20 @@ class FeedFragment : BaseListFragment<FeedState, Unit>() {
     override fun onViewCreated(rootView: View, savedInstanceState: Bundle?) {
         super.onViewCreated(rootView, savedInstanceState)
 
-        viewModel = ViewModelProviders.of(this, FeedViewModel.Factory(requireContext(), groupId)).get(FeedViewModel::class.java)
+        viewModel = ViewModelProviders
+                .of(this, FeedViewModel.Factory(requireContext(), groupId, showPlayedItems))
+                .get(FeedViewModel::class.java)
         viewModel.stateLiveData.observe(viewLifecycleOwner, Observer { it?.let(::handleResult) })
+
+        val shouldUseGridLayout = shouldUseGridLayout()
+        groupAdapter.spanCount = if (shouldUseGridLayout) getGridSpanCount() else 1
```

Comment:
Maybe this could be simplified to:
```suggestion
        groupAdapter.spanCount = if (shouldUseGridLayout()) getGridSpanCount() else 1
```